src/devices/controller.c

```diff
@@ -242,10 +241,8 @@ int32_t controller_print(controller_id_e_t id, uint8_t line, uint8_t col, const
 
 	va_list args;
 	va_start(args, fmt);
-	char* buf;
-	vasprintf(&buf, fmt, args);
-
-	buf[CONTROLLER_MAX_COLS] = '\0';
```

Comment:
we should still null-terminate the string. doesn't hurt, and may be safer

Reply:
`vsnprintf()` automatically appends a null terminator to the destination string so we're covered there.